bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/ValidateProjectEncoding.java

```diff
@@ -83,9 +84,10 @@ public IStatus runInWorkspace(IProgressMonitor monitor) {
 	static void updateMissingEncodingMarker(IProject project) {
 		try {
 			if (project.isAccessible() && !project.isHidden()) {
+				int severity = getSeverity();
 				String defaultCharset = getDefaultCharset(project);
-				if (defaultCharset == null) {
-					createEncodingMarker(project);
+				if (severity >= 0 && defaultCharset == null) {
```

Comment:
```severity != -1``` instead? Since the magic number is -1.

I think you can define the ```-1``` here as a constant and use it in platform UI.

The code in platform UI will complain about using non-API, but I think that is better than having the same magic number hard-coded in 2 places. Even if its unlikely for that magic number to change.

Reply:
Added constant and used it here.

Updated the UI integration (on the the UI repo) to use that constant too.
(However, that change doesn't feel good to me. Internal code in Eclipse plug-ins is internal to convey the information: "It's safe to alter things here w/o breaking contract for client plugins". But if this is the usual approach between the Eclipse platform bundles anyway, it might be another story. Still, this change wouldn't make things any better imho.)